raitools/raitools/_managers/counterfactual_manager.py

```diff
@@ -2,32 +2,203 @@
+        else:
+            self._counterfactual_config_list.append(new_counterfactual_config)
 
     def add(self,
-            continuous_features=None,
-            outcome_name=None,
-            total_CFs=None,
+            continuous_features,
+            total_CFs,
+            method='random',
+            desired_class=None,
             desired_range=None,
-            desired_class="opposite",
             permitted_range=None,
-            features_to_vary=None,
-            method=None):
-        raise NotImplementedError(
-            "Add not implemented for CounterfactualManager")
+            features_to_vary=None):
+        """Add a counterfactual generation configuration to be computed later.
+        """
+
+        counterfactual_config = _CounterfactualConfig(
+            method=method,
+            continuous_features=continuous_features,
+            total_CFs=total_CFs,
+            desired_class=desired_class,
+            desired_range=desired_range,
+            permitted_range=permitted_range,
+            features_to_vary=features_to_vary)
+
+        self._add_counterfactual_config(counterfactual_config)
 
     def compute(self):
-        pass
+        """Computes the counterfactual examples by running the counterfactual
+           configuration."""
+        for cf_config in self._counterfactual_config_list:
+            if not cf_config.is_computed:
+                cf_config.is_computed = True
+                try:
+                    dice_explainer = self._create_diceml_explainer(
+                        method=cf_config.method,
+                        continuous_features=cf_config.continuous_features)
 
-    def get(self):
-        raise NotImplementedError(
-            "Get not implemented for CounterfactualManager")
+                    X_test = self._test.drop([self._target_column], axis=1)
+
+                    counterfactual_obj = \
+                        dice_explainer.generate_counterfactuals(
+                            X_test, total_CFs=cf_config.total_CFs,
+                            desired_class=cf_config.desired_class,
+                            desired_range=cf_config.desired_range)
+
+                    cf_config.counterfactual_obj = \
+                        counterfactual_obj
+                except Exception as e:
+                    cf_config.has_computation_failed = True
+                    cf_config.failure_reason = str(e)
+
+    def get(self, failed_to_compute=False):
+        if not failed_to_compute:
+            counterfactual_obj_list = []
+            for counterfactual_config in self._counterfactual_config_list:
+                if counterfactual_config.is_computed:
+                    if not counterfactual_config.has_computation_failed:
+                        counterfactual_obj_list.append(
+                            counterfactual_config.counterfactual_obj)
+            return counterfactual_obj_list
+        else:
+            feature_reason_list = []
```

Comment:
confused about the name, should this be failure_reason_list instead of feature_reason_list?

Reply:
changed to failure_reason_list.